modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/IgniteLockBenchmark.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.yardstick.cache;
+
+import java.util.Map;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteLock;
+
+/**
+ * Ignite benchmark that performs Ignite.reentrantLock operations.
+ */
+public class IgniteLockBenchmark extends IgniteCacheLockBenchmark {
+    /** {@inheritDoc} */
+    @Override public boolean test(Map<Object, Object> map) throws Exception {
+        final int n = nextRandom(args.range());
+        final String key = "key" + n;
+        final String otherKey = "other_" + key;
+        final IgniteCache<String, Integer> cache = cacheForOperation();
+        final IgniteLock lock = ignite().reentrantLock(key, false, false, true);
+
+        lock.lock();
+
+        try {
+            cache.put(otherKey, cache.get(otherKey) + 1);
+        }
+        finally {
+            lock.unlock();
```

Comment:
we also need to close the reentrantLock after unlock, but now we have other problem, what if we close it and other thread will try to lock closed lock? please add something like:
`
IgniteLock lock;
while(true) {
  try{
     lock = ignite().reentrantLock(key, false, false, true);
     lock.lock();
   }catch (Exception e) {
      /*ignore or may be not?*/
      continue;
   }
   break;
}
`